azure/azure.go

```diff
@@ -114,6 +241,87 @@ func (a *azureOps) InstanceID() string {
 	return a.instance
 }
 
+func (a *azureOps) InspectInstance(instanceID string) (*cloudops.InstanceInfo, error) {
+
+	instInfo := &cloudops.InstanceInfo{
+		CloudResourceInfo: cloudops.CloudResourceInfo{
+			Name:   a.instance,
+			ID:     a.instance,
+			Zone:   "",
+			Region: "",
+		},
+	}
+	return instInfo, nil
+}
+
+func (a *azureOps) InspectInstanceGroupForInstance(instanceID string) (*cloudops.InstanceGroupInfo, error) {
+
+	ctx := context.Background()
+	agentPool, err := a.agentPoolsClient.Get(ctx, a.resourceGroupName, a.managedClusterName, a.agentPoolName)
+	if err != nil {
+		return nil, fmt.Errorf("Failed to get node pool details for [%s] of cluster [%v] in [%v] resource group."+
+			"Error: [%v] ", a.agentPoolName, a.managedClusterName, a.resourceGroupName, err)
+	}
+
+	// TODO: Currently multi-availability feature is in preview and
+	// zone field is not populated by azure agentPoolClient.
+	// For now, populate it with dummy value.
+	retval := &cloudops.InstanceGroupInfo{
+		CloudResourceInfo: cloudops.CloudResourceInfo{
+			ID:   *agentPool.ID,
+			Name: *agentPool.Name,
+		},
+		Zones: []string{"dummyzone"},
+	}
+
+	return retval, nil
+}
+
+func (a *azureOps) GetInstanceGroupSize(instanceGroupID string) (int64, error) {
+
+	ctx := context.Background()
+	agentPool, err := a.agentPoolsClient.Get(ctx, a.resourceGroupName, a.managedClusterName, instanceGroupID)
+	if err != nil {
+		return 0, err
+	}
+	return int64(*agentPool.Count), nil
```

Comment:
As this is a pointer, may be we should check that this is not null and return a default value with or without error in case it is nil.
The doc in the azure sdk says - 
```
// Count - Number of agents (VMs) to host docker containers. Allowed values must be in the range of 1 to 100 (inclusive). The default value is 1
```
@harsh-px any harm returning 1 if this is nil for some reason?

Reply:
what if we return count 0 with error if `*agentPool.Count` is nil? API consumer can check for err before consuming valume

Reply:
Just check the nil error as discussed and return any default value (0/1) as per your requirement in the consuming code.

Reply:
added check for `nil` pointer here